autoload/jedi.vim

```diff
@@ -190,6 +190,9 @@ function! jedi#_vim_exceptions(str, is_eval)
     try
         if a:is_eval
             let l:result.result = eval(a:str)
+            if type(l:result.result) != type('')
+                let l:result.result = string(l:result.result)
+            endif
         else
```

Comment:
The problem here is that we break list handling etc..

I think there's a regression somewhere else (e.g. old Neovim with new python-neovim) that is causing this.  See the discussion at #553.